python/paddle/fluid/layers/control_flow.py

```diff
@@ -1031,9 +1031,12 @@ def body(i, ten):
             output_vars = body(*loop_vars)
         if not isinstance(output_vars, (list, tuple)):
             output_vars = [output_vars]
-        if len(output_vars) != len(loop_vars):
-            raise ValueError("body in while_loop should return the same arity "
-                             "(length and structure) and types as loop_vars")
+        try:
+            assert_same_structure(output_vars, loop_vars, check_types=False)
+        except ValueError as e:
+            raise ValueError(
+                "body in while_loop should return the same arity "
+                "(length and structure) and types as loop_vars: {0}".format(e))
```

Comment:
Remove "and types" in the ValueError message because you set `check_types=False` when calling assert_same_structure.

However, it's not your fault. I found the old code was wrong here, too. :-)

Reply:
Thanks, fixed.